src/averageAges.js

```diff
@@ -15,11 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with logical operators (&&, ||) or ternary operator (?:)
-  // without nesting
+  return people.reduce((ageSum, man) => {
+    if (man.sex === 'm' && (century === undefined
+      || (man.died < century * 100 && man.died >= (century - 1) * 100))) {
+      return ageSum + man.died - man.born;
+    }
+
+    return ageSum;
+  }, 0) / people.filter(man => (man.sex === 'm' && (century === undefined
+    || (man.died < century * 100 && man.died >= (century - 1) * 100)))).length;
```

Comment:
you duplicate a lot of code, also use unneeded filter
Proposed solution for reduce only:
```javascript
people.reduce(({ sum, count }, person) => {
  if (...) {
    return { sum: sum + ..., count: count + 1 }
  }
  return { sum, count };
}, { sum: 0, count: 0 })
```

Your previous solution with external variable was better then current
Or you can just use for loop